graviti/dataset/dataset.py

```diff
@@ -243,6 +256,15 @@ def checkout(self, revision: str) -> None:
                 the branch, or the tag.
 
         """
+        try:
+            branch = self.branches.get(revision)
```

Comment:
what if the input revision is a tag?

Reply:
the revision in `self.commits.get(revision)` could be the tag name